src/main/java/com/cloudbees/jenkins/plugins/advisor/casc/AdvisorRootConfigurator.java

```diff
@@ -0,0 +1,151 @@
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Scalar;
+import io.jenkins.plugins.casc.model.Sequence;
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+/**
+ * A root element configurator used for configuring CloudBees Jenkins Advisor through {@link io.jenkins.plugins.casc.ConfigurationAsCode}
+ * https://github.com/jenkinsci/configuration-as-code-plugin
+ */
+@Extension(optional = true)
+@Restricted(NoExternalUse.class)
+public class AdvisorRootConfigurator extends BaseConfigurator<AdvisorGlobalConfiguration> implements RootElementConfigurator<AdvisorGlobalConfiguration> {
+
+    private static final Logger LOG = Logger.getLogger(AdvisorRootConfigurator.class.getName());
+
+    // Ignoring lastBundleResult since it is not configured in the plugin. It only informs about the last bundle generation
+    // Ignoring isValid since it is something auto-acalculated during the configuration
+    private final Collection<String> excludedAttributesInConf = Arrays.asList("lastBundleResult", "valid");
+
+    @NonNull
+    @Override
+    public String getName() {
+        return "advisor";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return getConfiguration().getDisplayName();
+    }
+
+    @Override
+    protected AdvisorGlobalConfiguration instance(Mapping mapping, ConfigurationContext configurationContext) throws ConfiguratorException {
+        // Scalar values
+        final String email = (mapping.get("email") != null ? mapping.getScalarValue("email") : StringUtils.EMPTY);
+        final String cc = (mapping.get("cc") != null ? mapping.getScalarValue("cc") : StringUtils.EMPTY);
+        final boolean nagDisabled = (mapping.get("nagDisabled") != null ? BooleanUtils.toBoolean(mapping.getScalarValue("nagDisabled")) : false);
+        final boolean acceptToS = (mapping.get("acceptToS") != null ? BooleanUtils.toBoolean(mapping.getScalarValue("acceptToS")) : false);
+
+        // List values
+        final Set<String> excludedComponents = new HashSet<>();
+        CNode excludedCN = mapping.get("excludedComponents");
+        if (excludedCN != null) {
+            if (excludedCN instanceof Sequence) {
+                Sequence s = (Sequence) excludedCN;
+                for (CNode cNode : s) {
+                    excludedComponents.add(cNode.asScalar().getValue());
+                }
+
+                if (excludedComponents.isEmpty()) {
+                    excludedComponents.add(AdvisorGlobalConfiguration.SEND_ALL_COMPONENTS);
+                }
+            } else {
+                throw new ConfiguratorException(this, "Excluded components are expected to be a collection.");
```

Comment:
```suggestion
                throw new ConfiguratorException(this, "Excluded components are expected to be a list.");
```

I think Yaml only has the concept of sequence/list?